internal/powerplatform/services/data_record/resource_data_record.go

```diff
@@ -0,0 +1,338 @@
+	DataRecordClient DataRecordClient
+	ProviderTypeName string
+	TypeName         string
+}
+
+type DataRecordResourceModel struct {
+	Id               types.String  `tfsdk:"id"`
+	EnvironmentId    types.String  `tfsdk:"environment_id"`
+	TableLogicalName types.String  `tfsdk:"table_logical_name"`
+	Columns          types.Dynamic `tfsdk:"columns"`
+}
+
+func (r *DataRecordResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + r.TypeName
+}
+
+func (r *DataRecordResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description:         "The Power Platform Data Record Resource allows the management of configuration records that are stored in Dataverse as records. This resource is not recommended for managing business data or other data that may be changed by Dataverse users in the context of normal business activities.",
+		MarkdownDescription: "The Power Platform Data Record Resource allows the management of configuration records that are stored in Dataverse as records. This resource is not recommended for managing business data or other data that may be changed by Dataverse users in the context of normal business activities.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Unique id (guid)",
+				Description:         "Unique id (guid)",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"environment_id": schema.StringAttribute{
+				Description: "Id of the Dynamics 365 environment",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"table_logical_name": schema.StringAttribute{
+				Description: "Logical name of the data record table",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"columns": schema.DynamicAttribute{
+				Description: "Columns of the data record table",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func (r *DataRecordResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	clientApi := req.ProviderData.(*api.ProviderClient).Api
+	if clientApi == nil {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+	r.DataRecordClient = NewDataRecordClient(clientApi)
+}
+
+func (r *DataRecordResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state *DataRecordResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
```

Comment:
so you are telling that you save state to "plan" var and plan to "state" var :D?

Reply:
renamed plan variables